apps/test/unit/code-studio/pd/application_dashboard/detail_view_contentsTest.js

```diff
@@ -349,35 +378,55 @@ describe('DetailViewContents', () => {
         expect(detailView.find('textarea#notes').prop('disabled')).to.be.true;
         expect(detailView.find('textarea#notes_2').prop('disabled')).to.be.true;
       });
+    });
+  }
 
-      it(`cannot make status incomplete from dropdown in ${applicationType}`, () => {
-        const detailView = mountDetailView(applicationType);
-        expect(
-          detailView
-            .find('#DetailViewHeader select')
-            .find('option')
-            .find('[value="incomplete"]')
-        ).to.have.lengthOf(0);
+  describe('Principal Approvals', () => {
+    it(`Shows principal responses if approval is complete`, () => {
+      const detailView = mountDetailView('Teacher');
+      expect(detailView.text()).to.contain(
+        'Principal Approval and School Information'
+      );
+    });
+    it(`Shows URL to principal approval if sent and incomplete`, () => {
+      const guid = '1020304';
+      const detailView = mountDetailView('Teacher', {
+        applicationData: {
+          ...DEFAULT_APPLICATION_DATA,
+          application_guid: guid,
+          principal_approval_state: 'Incomplete'
+        }
       });
-
-      it(`incomplete status is in dropdown if ${applicationType} application is incomplete`, () => {
-        const detailView = mountDetailView('Teacher', {
-          applicationData: {
-            ...DEFAULT_APPLICATION_DATA,
-            status: 'incomplete',
-            scholarship_status: null,
-            update_emails_sent_by_system: false
-          }
-        });
-        expect(
-          detailView
-            .find('#DetailViewHeader select')
-            .find('option')
-            .find('[value="incomplete"]')
-        ).to.have.lengthOf(1);
+      expect(
+        detailView
+          .find('a')
+          .last()
+          .props().href
+      ).to.contain(`principal_approval/${guid}`);
+    });
+    it(`Shows button to make principal approval required if not`, () => {
+      const detailView = mountDetailView('Teacher', {
+        applicationData: {
+          ...DEFAULT_APPLICATION_DATA,
+          principal_approval_state: 'Not required'
+        }
       });
+      expect(detailView.find('PrincipalApprovalButtons').text()).to.contain(
+        'Make required'
+      );
     });
-  }
+    it(`Shows button to make principal approval not required if it is`, () => {
```

Comment:
I went back and forth on how thoroughly to test these. I could ensure that once this button is clicked, the right things happen –– the right data gets sent, we set state correctly. But testing state changes isn't good (instead, I'd test that the new button shows up, from Make required to Make not required or vice-versa), and we aren't testing other ajax calls in this test suite. But happy for pushback here –– I went with the easiest, which isn't always the best ;)

Reply:
all makes sense. my understanding is that this is an area where we have decided that we do want to invest in better test coverage in code we are touching. Based on that, I do think it would be worth making adding a bit more test coverage now. this could either be the kind of test you landed on above, or UI tests (they would look a lot like your screen recordings). since the principal approval button appears in a couple different places, I think the better bang-for-the-buck would be the kind of test you were talking about:

> test that the new button shows up, from Make required to Make not required or vice-versa

there should be examples of other tests doing something similar using sinon to mock the ajax requests to provide server responses. let me know if you want help identifying a good example to follow.

Reply:
quick follow-up: I guess you would still need a separate test for each view regardless of whether you are using a js test or a UI test 🤷‍♂️ my suggestion would still be to focus on improving the js tests to cover the new behavior. UI tests could be a nice bonus.

Reply:
Sounds good –– I think mocking the ajax requests makes sense. I did that for the teacher apps, so I'm guessing it'll look similar.